samples/template-operator/operator/controllers/sample_controller.go

```diff
@@ -47,188 +39,16 @@ const deletionFinalizer = "deletion-finalizer"
-		sampleResource.Status.State = v1alpha1.SampleStateError
-		return r.Client.Status().Update(ctx, sampleResource)
-	}
-
-	// Use manifest library client to install a sample chart
-	installInfo, err := r.PrepareInstallInfo(ctx, sampleResource)
-	if err != nil {
-		return err
-	}
-
-	ready, err := manifestClient.Install(installInfo)
-	if err != nil {
-		sampleResource.Status.State = v1alpha1.SampleStateError
-		return r.Client.Status().Update(ctx, sampleResource)
-	}
-	if ready {
-		sampleResource.Status.State = v1alpha1.SampleStateReady
-		return r.Client.Status().Update(ctx, sampleResource)
-	}
-	return nil
-}
-
-func (r *SampleReconciler) getManifestClient(logger *logr.Logger, configString string,
-) (*manifestLib.Operations, error) {
-	config := map[string]interface{}{}
-	if err := strvals.ParseInto(configString, config); err != nil {
-		return nil, err
-	}
-	// Example: Prepare manifest library client
-	return manifestLib.NewOperations(logger, r.Config, "nginx-release-name", cli.New(),
-		map[string]map[string]interface{}{
-			// check --set flags parameter for helm
-			"set": {},
-			// comma separated values of manifest command line flags
-			"flags": config,
-		})
-}
-
-func (r *SampleReconciler) HandleDeletingState(ctx context.Context, sampleResource *v1alpha1.Sample) error {
-	// TODO: deletion logic here
-	logger := log.FromContext(ctx)
-	manifestClient, err := r.getManifestClient(&logger, sampleResource.Spec.ChartFlags)
-	if err != nil {
-		sampleResource.Status.State = v1alpha1.SampleStateError
-		return r.Client.Status().Update(ctx, sampleResource)
-	}
-
-	// Use manifest library client to install a sample chart
-	installInfo, err := r.PrepareInstallInfo(ctx, sampleResource)
-	if err != nil {
-		return err
-	}
-
-	readyToBeDeleted, err := manifestClient.Uninstall(installInfo)
-	if err != nil {
-		sampleResource.Status.State = v1alpha1.SampleStateError
-		return r.Client.Status().Update(ctx, sampleResource)
-	}
-	if readyToBeDeleted {
-		// Example: If Deleting state, remove Finalizers
-		sampleResource.Finalizers = nil
-		return r.Client.Update(ctx, sampleResource)
-	}
-	return nil
-}
-
-func (r *SampleReconciler) HandleErrorState(ctx context.Context, sampleResource *v1alpha1.Sample) error {
-	// TODO: error logic here
-
-	// Example: If Error state, set state to Processing
-	sampleResource.Status.State = v1alpha1.SampleStateProcessing
-	return r.Client.Status().Update(ctx, sampleResource)
-}
-
-func (r *SampleReconciler) HandleReadyState(_ context.Context, _ *v1alpha1.Sample) error {
-	// TODO: ready logic here
-
-	// Example: If Ready state, check consistency of deployed module
-	return nil
-}
-
 // SetupWithManager sets up the controller with the Manager.
 func (r *SampleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.Config = mgr.GetConfig()
+	if err := r.Inject(mgr, &v1alpha1.Sample{},
+		declarative.WithResourceLabels(
+			map[string]string{"sampleKey": "sampleValue"},
+		)); err != nil {
+		return err
```

Comment:
err check not needed

Reply:
Why?

Reply:
Actually not sure anymore, sorry for the confusion